schematic/store/synapse.py

```diff
@@ -1,10 +1,12 @@
+from datetime import datetime, timedelta
 from copy import deepcopy
 import os
 import uuid  # used to generate unique names for entities
 import json
 import atexit
 import logging
 import secrets
+from dataclasses import dataclass
```

Comment:
Do you think it would be useful in the future to use pydantic's `dataclass` for the validation aspect?

Reply:
I have not yet looked into the details of dydantic's dataclass. But that's something we could consider for sure.